Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -836,19 +836,26 @@ public final class PackageBuilder {
     }
 
     /// Returns the list of platforms supported by the manifest.
-    func platforms() -> [SupportedPlatform] {
-        if let platforms = _platforms {
+    func platforms(isTest: Bool = false) -> [SupportedPlatform] {
+        if let platforms = _platforms[isTest] {
             return platforms
         }
 
         var supportedPlatforms: [SupportedPlatform] = []
 
         /// Add each declared platform to the supported platforms list.
         for platform in manifest.platforms {
+            let declaredPlatform = platformRegistry.platformByName[platform.platformName]!
+            var version = PlatformVersion(platform.version)
+
+            // macOS tests have a higher minimum deployment target
+            if declaredPlatform == Platform.macOS && isTest && version < PlatformVersion("10.15") {
```

Comment:
This isn't future proof. What we actually want here is whatever the deployment target of XCTest is. What I would suggest is to read the LC_BUILD_VERSION load command from the XCTest framework, and use that.

I don't remember if we have any Mach-O parsing utilities in SwiftPM or TSC, but if not, you can read it using vtool and parse the output.

```
$ xcrun vtool -show-build $(xcrun --sdk macosx --show-sdk-platform-path)/Developer/Library/Frameworks/XCTest.framework/XCTest
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/XCTest.framework/XCTest:
Load command 8
      cmd LC_BUILD_VERSION
  cmdsize 32
 platform MACOS
    minos 10.14
      sdk 10.15.4
   ntools 1
     tool LD
  version 556.6
Load command 9
      cmd LC_BUILD_VERSION
  cmdsize 32
 platform MACCATALYST
    minos 13.1
      sdk 13.4
   ntools 1
     tool LD
  version 556.6

```

Just grab the value for the first line starting with "minos".